src/ImageSharp/Image{TPixel}.cs

```diff
@@ -104,6 +139,11 @@ internal Image(Configuration configuration, ImageMetaData metadata, IEnumerable<
         /// </summary>
         public IImageFrameCollection<TPixel> Frames => this.frames;
 
+        /// <summary>
+        /// Gets the clear color to initialize the image frame pixels with.
+        /// </summary>
+        internal TPixel? ClearColor => this.clearColor;
```

Comment:
Not sure if I like this.

Does it worth to pollute our object model in this manner just to make it easier to add frames of a "default" color when producing gif-s manually? (0.01% of ImageSharp use-cases I guess).
@JimBobSquarePants @tocsoft thoughts?

Reply:
I'm not a fan of nullable type properties on classes. I'm voting no.

Reply:
The thought behind it is that if you create an image with a background color, all its frames will have automatically that background color, which could be reasonably expected from a user stand point. Without it you'd have the first frame with the background color, and if the user adds another frame with ImageFrameCollection.CreateFrame(), it will also have that background color.

What's the reason for not having nullable type properties on a class?

Reply:
if we have this atall you would want to distinguish between default and not set because default(TPixel) would be == Transparent and you want to skip the code path if its not set at all.

Reply:
Transparent isn’t the same as default. Rgb = 255.

Reply:
oh yeah.. your right, ignore that in that case we can just skip the logic on `default(TPixel)` 👍

Reply:
Agreed with the `default(TPixel)` part, so you suggest always initializing the image frame with the clearColor then instead of using `memoryManager.AllocateClean2D`?

I don't agree with the nullable reasoning, if it's properly documented and default values are not illogical I don't see a reason to refrain from using certain language features, if TPixel were a class it'd have been the same. In this case non-nullable works as well, but in general if nullable is better suitable for a particular situation I wouldn't handicap the design/api out of fear users are not capable enough of using it.

Reply:
Alright, committed the change to non-nullable clear color.

Reply:
It wasn't a question of handicapping the API nor one of fear. Nullable properties have their place, just not in any API where you hold all the cards.

Take for example the new `ResponseHeaders` class from `Microsoft.Http.Abstractions`. The `ContentLength` property there is of type `long?` this makes perfect sense since the `HttpResponse` might not provide a content-length header.  

We have absolute control over that property and assigning it so Nullable would have been the incorrect option design wise.

On that note... I wonder whether this should be a property of the `ImageFrame<T>` itself?  That way we base any non-background color assigned frames on the root frame which would keep the property in line with the Width/Height properties.

Reply:
I've moved it to ImageFrame, it's indeed more consistent with the Width/Height properties.